tests/ComponentTest.php

```diff
@@ -25,6 +27,29 @@ public function tearDown()
         parent::tearDown();
     }
 
+    public function testComponentLookup()
+    {
+        $service = new ComponentService();
+        $service->componentTemplatePaths = [
+            'components/subfolder'
+        ];
+
+        try {
+            $resultHTML = $service->renderComponent('SubComponent', []);
+        } catch (\Exception $e) {
+            $this->assertTrue(strpos($e->getMessage(), "None of the following templates could be found") !== false);
+        }
+
+        $service->componentTemplatePaths = [
+            'components/subfolder'
+        ];
+
+        $resultHTML = $service->renderComponent('SubComponent', [], null);
+
+        $expected = '<button class="subcomponent">Text</button>';
+        $this->assertEquals($expected, trim($resultHTML->getValue()));
+    }
+
```

Comment:
This test should follow the patterns used throughout this file.

- `testCustomizedComponentPath` is preferred
- You should be modifying the Configs then just attempting to render `<:SubComponent/>`. As stated in a seperate comment on this PR, I'd rather the Injector be using the `private static` configs as they're less verbose / less error prone in my experience. However, if we need the flexability of the Injector for other frameworks, you can use `Injector::inst()->load()` to update configs on the fly.

Reply:
As a test though, you should be isolating the tested code from other moving parts, hence creating the `$service = new ComponentService();` directly and setting the dependency that the method uses explicitly, rather than the possibility that unrelated system state may change the meanings of these variables. For example, executing this test in context of a project that already had the `componentTemplatePaths` configured for `components/atoms` may cause it to succeed when it should have failed.

Reply:
The assumption is that this test would only run in isolation, not within a project. (ie. on Travis only)
So that wouldn't happen right?

Reply:
But people _writing_ tests do run them in context of projects, which can lead to unexpected issues - and if your project heavily relies on a particular module, you should probably be running that module's tests alongside the project. 

Based on your other note about preferring `Config`, this code will need to change anyway so will do that.

Reply:
If you want to add/update unit tests for this module, you should be running them against the same thing that Travis CI does, not your personal project IMO. (ie. an empty SS project)